examples/benchmark/py/perf_run.py

```diff
@@ -0,0 +1,301 @@
+    # Warm up
+    with torch.no_grad():
+        for _ in range(WARMUP_ITER):
+            features = model(*input_tensors)
+
+    torch.cuda.synchronize()
+
+    timings = []
+    with torch.no_grad():
+        for i in range(iters):
+            start_time = timeit.default_timer()
+            features = model(*input_tensors)
+            torch.cuda.synchronize()
+            end_time = timeit.default_timer()
+            meas_time = end_time - start_time
+            timings.append(meas_time)
+            print("Iteration {}: {:.6f} s".format(i, end_time - start_time))
+    
+    printStats("Torch-TensorRT", timings, precision)
+
+def torch_dtype_from_trt(dtype):
+    if dtype == trt.int8:
+        return torch.int8
+    elif dtype == trt.bool:
+        return torch.bool
+    elif dtype == trt.int32:
+        return torch.int32
+    elif dtype == trt.float16:
+        return torch.float16
+    elif dtype == trt.float32:
+        return torch.float32
+    else:
+        raise TypeError("%s is not supported by torch" % dtype)
+
+def torch_device_from_trt(device):
+    if device == trt.TensorLocation.DEVICE:
+        return torch.device("cuda")
+    elif device == trt.TensorLocation.HOST:
+        return torch.device("cpu")
+    else:
+        return TypeError("%s is not supported by torch" % device)
+
+
+def run_tensorrt(model, input_tensors, params, precision, is_trt_engine=False):
+    engine = None
+
+    # If the model file is a TensorRT engine then directly deserialize and run inference
+    # else convert the torch module to a TensorRT engine first and then run inference
+    if not is_trt_engine:
+        compile_settings = {
+            "inputs": input_tensors,
+            "enabled_precisions": {precision_to_dtype(precision)}
+        }
+
+        print("Converting method to TensorRT engine...")
+        with torch.no_grad():
+            model = torchtrt.ts.convert_method_to_trt_engine(model, "forward", **compile_settings)
+    
+    # Deserialize the TensorRT engine
+    with trt.Logger() as logger, trt.Runtime(logger) as runtime:
+        engine = runtime.deserialize_cuda_engine(model)
+  
+    print("Running TensorRT")
+    iters = params.get('iterations', 20)
+    batch_size = params.get('batch', 1)
+
+    # Compiling the bindings
+    bindings = engine.num_bindings * [None]
+
+    k = 0
+    for idx,_ in enumerate(bindings):
+        dtype = torch_dtype_from_trt(engine.get_binding_dtype(idx))
+        shape = (batch_size,) + tuple(engine.get_binding_shape(idx))
+        device = torch_device_from_trt(engine.get_location(idx))
+        if not engine.binding_is_input(idx):
+            # Output bindings
+            output = torch.empty(size=shape, dtype=dtype, device=device)
+            bindings[idx] = output.data_ptr()
+        else:
+            # Input bindings
+            bindings[idx] = input_tensors[k].data_ptr()
+            k += 1
+    
+    timings = []
+    with torch.no_grad():
```

Comment:
torch.cuda.synchronize is required because we are using execute_async API and using the default CUDA stream in the process.

I got rid of torch.no_grad in this case